rmqproducer/rabbitmq_producer.py

```diff
@@ -0,0 +1,376 @@
+        confirmations on the channel. The only way to turn this off is to close
+        the channel and create a new one.
+
+        When the message is confirmed from RabbitMQ, the
+        on_delivery_confirmation method will be invoked passing in a Basic.Ack
+        or Basic.Nack method from RabbitMQ that will indicate which messages it
+        is confirming or rejecting.
+
+        """
+        self._LOGGER.info('Issuing Confirm.Select RPC command')
+        self._channel.confirm_delivery(self.on_delivery_confirmation)
+        self._connection.ioloop.stop()
+
+    def on_delivery_confirmation(self, method_frame):
+        """Invoked by pika when RabbitMQ responds to a Basic.Publish RPC
+        command, passing in either a Basic.Ack or Basic.Nack frame with
+        the delivery tag of the message that was published. 
+
+        The delivery tag is an integer counter indicating the message number 
+        that was sent on the channel via Basic.Publish. Here we're just doing 
+        house keeping to keep track of stats and remove message numbers that 
+        we expect a delivery confirmation of from the list used to keep track 
+        of messages that are pending confirmation.
+
+        :param pika.frame.Method method_frame: Basic.Ack or Basic.Nack frame
+
+        """
+        confirmation_type = method_frame.method.NAME.split('.')[1].lower()
+        message_num = method_frame.method.delivery_tag
+        if confirmation_type == 'ack':
+            self._LOGGER.info('Message %i published successfully',
+                              message_num)
+            self._LOGGER.debug('The message published: %s',
+                               self._messages[message_num])
+        if confirmation_type == 'nack':
+            self._LOGGER.error('The message %i failed to publish: %s',
+                               message_num, self._messages[message_num])
+            if self.nack_callback:
+                self.nack_callback(self._messages[message_num])
+        try:
+            del self._messages[message_num]
+        except KeyError:
+            self._LOGGER.error(
+                "The message to delete after confirmation not present in dictionary")
+        self._connection.ioloop.stop()
+
+    def publish_message(self, message, routing_key):
+        """If the class is not stopping, publish a message to RabbitMQ,
+        appending a list of deliveries with the message number that was sent.
+        This list will be used to check for delivery confirmations in the
+        on_delivery_confirmations method.
+
+        :param str message: The message to be published
+        :param str routing_key: The routing key for the message to be published
+
+        """
+        if self._channel_closing:
+            self._LOGGER.error(
+                "Following message couldn't be published: %s", message)
+            return
+        self._channel.basic_publish(self.exchange, routing_key, message)
```

Comment:
As per the research done by @kgprocks there is no way to pass the delivery_tag with the message. If you get across some way, please let us know.